neo4j/driver_with_context.go

```
@@ -48,6 +50,48 @@ const (
+	//		config.Database = "my-db"
+	//		config.RoutingControl = neo4j.Writers
+	//	})
+	// ExecuteQuery causal consistency is guaranteed by default across different successful calls to ExecuteQuery
+	// targeting the same database.
+	// In other words, a successful read query run by ExecuteQuery is guaranteed to be able to read results created
+	// from a previous successful write query run by ExecuteQuery on the same database.
+	// This is achieved through the use of a bookmarks, managed by a default neo4j.BookmarkManager instance.
+	// This default BookmarkManager instance can be retrieved with GetDefaultManagedBookmarkManager.
+	// Such a consistency guarantee is not maintained between ExecuteQuery calls and the lower-level
+	// neo4j.SessionWithContext API calls, unless sessions are explicitly configured with the same bookmark manager.
+	// That guarantee may also break if a custom implementation of neo4j.BookmarkManager is provided via for instance
+	// the built-in callback neo4j.WithBookmarkManager.
+	// You can disable bookmark management with neo4j.WithoutBookmarkManager.
+	ExecuteQuery(context.Context, string, map[string]any, ...ExecuteQueryConfigurationOption) (*EagerResult, error)
+	// GetDefaultManagedBookmarkManager returns the bookmark manager instance used by ExecuteQuery by default.
+	// This is useful when ExecuteQuery is called without custom bookmark managers and the lower-level
+	// neo4j.SessionWithContext APIs are called as well.
+	// In that case, the recommended approach is as follows:
+	// 	results, err := driver.ExecuteQuery(ctx, query, params)
+	// 	// [...] do something with results and error
+	//	bookmarkManager := driver.GetDefaultManagedBookmarkManager()
+	// 	// maintain consistency with sessions as well
+	//	session := driver.NewSession(ctx, neo4j.SessionConfig {BookmarkManager: bookmarkManager})
+	//	// [...] run something within the session
+	GetDefaultManagedBookmarkManager() BookmarkManager
```

Interestingly enough, the ADR doesn't specify the name of this getter. Java, JS, and Python went with `queryBookmarkManager`, while .NET seems to not expose it for now. Before merging any of the languages, we should sit together one more time and discuss the naming.

---

Here are some name schemes I brainstormed
* `BabysFirstBookmarkManager`
  * Together with renaming `executeQuery` to `BabiesFirstQueryExecutor`
* /joking
* `ManagedBookmarkManager`
* `QueryBookmarkManager`
* `ExecuteQueryBookmarkManager`
* `Eager(Result)BookmarkManager`
* `Auto(matic)BookmarkManager`
* `DriverLevelBookmarkManager` (my current favorite, if we decide to use it by default for all driver-level execution APIs)
* `DefaultBookmarkManager` (as the ADR says, misleading: it's not being used for sessions by default)
* `SimpleApiBookmarkManager`
* `HighLevelBookmarkManager`

---

BasicBookmarkManager

---

> `GlobalBookmarkManager`

I argue this is also misleading as it's not the same BMM across multiple drivers.

> `StaticBookmarkManager`

I like that one even though it's not really static in the Java sense.

So far I don't find any one to stand out. There are just some that I def. don't like.

---

Picked `DefaultExecuteQueryBookmarkManager` for now (see c03dea4c04433046f66b5f600bc89b730b93e964)